src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt

```diff
@@ -72,6 +76,7 @@ class ParticipantDataIterable(private val supplier: NeighborPageSupplier) : Iter
         private var finishedCurrentPage = false
         private var currentIndex = 0
         private var currentPage: NeighborPage? = null
+        private var isFirstPage = true
```

Comment:
This likely needs to be synchronized/have added threadsafety

Reply:
I don't see why we need to though. can you expound?